### PR TITLE
DeformableConvolution v8: MO and IR reader

### PR DESCRIPTION
**Pull request by itikhono**
### Details:
 - *ModelOptimizer part*
 - *IR reader tests*

### Tickets:
 - *56576, 56582*

### CODE REVIEW

File: inference-engine/tests/functional/inference_engine/ngraph_reader/deformable_convolution_opset8_tests.cpp [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+				<port id="0" precision="FP32">
+					<dim>256</dim>
+					<dim>256</dim>
+					<dim>3</dim>
+					<dim>3</dim>
+				</port>
+			</output>
+		</layer>
+	    <layer id="3" name="1090" type="DeformableConvolution" version="opset8">
+            <data deformable_group="1" dilations="1,1" group="1" pads_begin="1,1" pads_end="1,1" strides="2,2"/>
```
Comment (ilya-lavrenov) [inline inference-engine/tests/functional/inference_engine/ngraph_reader/deformable_convolution_opset8_tests.cpp:46]: @ilyachur do we need such test? I suppose we agreed to have only visitor api tests for operations

Comment (ilyachur) [reply inference-engine/tests/functional/inference_engine/ngraph_reader/deformable_convolution_opset8_tests.cpp:46]: @itikhono Yes, we don't need to add new `ngraph_reader` tests. visitor tests should cover this functionality

File: ngraph/core/include/ngraph/op/deformable_convolution.hpp [hunk 2]
```diff
@@ -171,7 +171,7 @@ namespace ngraph
                 }
 
             private:
-                int64_t m_bilinear_interpolation_pad;
+                bool m_bilinear_interpolation_pad;
```
Comment (ilyachur) [inline ngraph/core/include/ngraph/op/deformable_convolution.hpp:174]: Why did you change int64 to bool?

Comment (itikhono) [reply ngraph/core/include/ngraph/op/deformable_convolution.hpp:174]: according to spec, it was a mistake

File: inference-engine/tests/functional/inference_engine/ngraph_reader/deformable_convolution_opset8_tests.cpp (hunk 1 above)
Comment (itikhono) [reply inference-engine/tests/functional/inference_engine/ngraph_reader/deformable_convolution_opset8_tests.cpp]: the tests have been deleted

File: model-optimizer/extensions/front/mxnet/modulated_deformable_conv_ext.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    enabled = True
+
+    @classmethod
+    def extract(cls, node):
+        attr = get_mxnet_layer_attrs(node.symbol_dict)
+
+        kernel = attr.tuple("kernel", int, None)
+        stride = attr.tuple("stride", int, tuple(np.ones(len(kernel), dtype=np.int64)))
+        padding = attr.tuple("pad", int, tuple(np.zeros(len(kernel), dtype=np.int64)))
+        dilate = attr.tuple("dilate", int, tuple(np.ones(len(kernel), dtype=np.int64)))
```
Comment (rkazants) [inline model-optimizer/extensions/front/mxnet/modulated_deformable_conv_ext.py:22]: Please fix specification because I see that default value for all these attributes is `None`.

File: model-optimizer/mo/ops/convolution.py [hunk 4]
```diff
@@ -237,7 +236,7 @@ def infer(node: Node):
         for n in node.out_nodes():
             node.out_node(n).shape = output_shape
 
-        mark_input_bins(node, start_port=1 if node.type != 'DeformableConvolution' else 2)
+        #mark_input_bins(node, start_port=1 if node.type != 'DeformableConvolution' else 2)
```
Comment (rkazants) [inline model-optimizer/mo/ops/convolution.py:239]: why is this commented out? It can affect `Convolution` operation.

File: model-optimizer/extensions/front/mxnet/modulated_deformable_conv_replacer.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from mo.front.common.replacement import FrontReplacementPattern
+from mo.graph.graph import Graph
+
+
+class DeformableConvolutionReplacer(FrontReplacementPattern):
+    # swap mask and weights inputs for ModulatedDeformableConvolution accroding to the specification
```
Comment (rkazants) [inline model-optimizer/extensions/front/mxnet/modulated_deformable_conv_replacer.py:9]: according

File: model-optimizer/extensions/front/mxnet/modulated_deformable_conv_replacer.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    enabled = True
+
+    def find_and_replace_pattern(self, graph: Graph):
+        print(graph.get_op_nodes())
```
Comment (rkazants) [inline model-optimizer/extensions/front/mxnet/modulated_deformable_conv_replacer.py:13]: remove `print`

File: model-optimizer/mo/ops/convolution.py (hunk 4 above)
Comment (itikhono) [reply model-optimizer/mo/ops/convolution.py:239]: could you explain why we should mark some inputs as bin?

File: model-optimizer/mo/ops/deformable_convolution.py [hunk 7]
```diff
@@ -13,16 +13,18 @@ def __init__(self, graph: Graph, attrs: dict):
         super().__init__(graph, {
             'type': __class__.op,
             'op': __class__.op,
-            'version': 'opset1',
+            'version': 'opset8',
             'infer': Convolution.infer,
             'group': 1,
             'deformable_group': 1,
             'multiplication_transparent': True,
             'multiplication_transparent_ports': [(0, 0), (2, 0)],
```
Comment (rkazants) [inline model-optimizer/mo/ops/deformable_convolution.py:21]: I can't find these two attributes in the specification. Please explain why they are needed in a comment.

File: model-optimizer/extensions/front/mxnet/modulated_deformable_conv_replacer.py (hunk 6 above)
Comment (itikhono) [reply model-optimizer/extensions/front/mxnet/modulated_deformable_conv_replacer.py:13]: nice catch, thanks!

File: model-optimizer/mo/ops/deformable_convolution.py (hunk 7 above)
Comment (itikhono) [reply model-optimizer/mo/ops/deformable_convolution.py:21]: these attributes are marked as required in the spec, so we have to provide some values when extracting them from different frameworks, it can be different values for different frameworks. For required attributes we don't have default values.

Comment (itikhono) [reply model-optimizer/mo/ops/deformable_convolution.py:21]: I'm not sure about these attributes, it is related to some preprocessing in MO, more precisely with BinarizeWeightsM1P1 transformation.

**Review by ilyachur — APPROVED** [review]
nGraph part LGTM

File: model-optimizer/extensions/front/mxnet/modulated_deformable_conv_replacer.py (hunk 6 above)
Comment (itikhono) [reply model-optimizer/extensions/front/mxnet/modulated_deformable_conv_replacer.py]: fixed

File: model-optimizer/extensions/front/mxnet/modulated_deformable_conv_replacer.py (hunk 5 above)
Comment (itikhono) [reply model-optimizer/extensions/front/mxnet/modulated_deformable_conv_replacer.py]: fixed

File: model-optimizer/mo/ops/convolution.py (hunk 4 above)
Comment (rkazants) [reply model-optimizer/mo/ops/convolution.py:239]: I propose to stay `mark_input_bins` because we have multiple transformations that use `bin` attribute. Or to remove it at all and arrange DL benchmark validation to see any regressions.

File: model-optimizer/mo/ops/deformable_convolution.py (hunk 7 above)
Comment (rkazants) [reply model-optimizer/mo/ops/deformable_convolution.py:21]: I see your point. It makes sense to have it.

File: model-optimizer/mo/ops/convolution.py (hunk 4 above)
Comment (itikhono) [reply model-optimizer/mo/ops/convolution.py]: reverted

**Review by sadolini — APPROVED** [review]

**Review by rkazants — APPROVED** [review]

File: model-optimizer/mo/ops/deformable_convolution.py [hunk 8]
```diff
@@ -13,16 +13,18 @@ def __init__(self, graph: Graph, attrs: dict):
...
             'out_ports_count': 1,
+            'bilinear_interpolation_pad': False,
         }, attrs)
 
     def backend_attrs(self):
-        # the same attributes as in a regular convolution and one additional attribute 'deformable_group' and 'group'
-        return Convolution(self.graph, {}).backend_attrs() + ['deformable_group', 'group']
+        # the same attributes as in a regular convolution and additional attributes 'deformable_group', 'group'
+        # and 'bilinear_interpolation_pad'
+        return Convolution(self.graph, {}).backend_attrs() + ['deformable_group', 'group', 'bilinear_interpolation_pad']
```
Comment (lazarevevgeny) [inline model-optimizer/mo/ops/deformable_convolution.py:30]: Here is the bug introduced. For the "opset1" version of the operation we should not add `bilinear_interpolation_pad` attribute. Implement a solution like in the `model-optimizer/extensions/ops/topk.py`

Comment (itikhono) [reply model-optimizer/mo/ops/deformable_convolution.py:30]: fixed in https://github.com/openvinotoolkit/openvino/pull/6806
